cmd/admin-trace.go

```diff
@@ -154,30 +152,36 @@ type responseInfo struct {
 	StatusCode int               `json:"statuscode,omitempty"`
 }
 type trace struct {
-	NodeName     string       `json:"nodename"`
-	FuncName     string       `json:"funcname"`
+	NodeName     string       `json:"host"`
+	FuncName     string       `json:"api"`
 	RequestInfo  requestInfo  `json:"request"`
 	ResponseInfo responseInfo `json:"response"`
 }
 
-// parse Operation name from function name
-func getOpName(fname string) (op string) {
-	res := funcNameRegex.FindStringSubmatch(fname)
-	op = fname
-	if len(res) == 2 {
-		op = res[1]
-	}
-	return
+// getOpName sanitizes the operation name for mc
+func getOpName(name string) (op string) {
+	op = strings.TrimPrefix(name, "github.com/minio/minio/cmd.")
+	op = strings.TrimSuffix(op, "Handler-fm")
+	op = strings.Replace(op, "objectAPIHandlers", "s3", 1)
+	op = strings.Replace(op, "webAPIHandlers", "s3", 1)
+	op = strings.Replace(op, "adminAPIHandlers", "admin", 1)
+	op = strings.Replace(op, "(*storageRESTServer)", "internal", 1)
+	op = strings.Replace(op, "(*peerRESTServer)", "internal", 1)
+	op = strings.Replace(op, "(*lockRESTServer)", "internal", 1)
+	op = strings.Replace(op, "stsAPIHandlers", "sts", 1)
+	op = strings.Replace(op, "LivenessCheckHandler", "healthcheck", 1)
+	op = strings.Replace(op, "ReadinessCheckHandler", "healthcheck", 1)
```

Comment:
This type of information should come from the server we shouldn't have hardcoded entries like this in `mc` related to MinIO server. Bring these constants as part of `pkg/madmin` and move these functions there instead.